src/benchmarks/micro/libraries/System.Runtime/Perf.String.cs

```diff
@@ -154,13 +154,20 @@ public string TrimEnd_CharArr(string s, char[] c)
             => s.TrimEnd(c);
 
         [Benchmark]
-        [Arguments("Hello", 'l', '!')] // Contains two 'l'
-        [Arguments("Hello", 'a', 'b')] // Contains one 'a'
-        [Arguments("This is a very nice sentence", 'z', 'y')] // 'z' does not exist in the string
-        [Arguments("This is a very nice sentence", 'i', 'I')] // 'i' occuress 3 times in the string
+        [ArgumentsSource(nameof(ReplaceArguments))]
         public string Replace_Char(string text, char oldChar, char newChar)
             => text.Replace(oldChar, newChar);
 
+        public static IEnumerable<object[]> ReplaceArguments()
+        {
+            yield return new object[] { "Hello", 'l', '!' };    // Contains two 'l'
+            yield return new object[] { "Hello", 'a', 'b' };    // Contains one 'a'
+            yield return new object[] { "This is a very nice sentence", 'z', 'y' }; // 'z' does not exist in the string
+            yield return new object[] { "This is a very nice sentence", 'i', 'I' }; // 'i' occuress 3 times in the string
+            yield return new object[] { new string('-', 100), '-', '+' };
+            yield return new object[] { new string('-', 1000), '-', '+' };
```

Comment:
Just picked random larger numbers.

Reply:
Would it be more realistic to not replace all the chars in the string? And also have them in varying offsets. Eg. something like this (making it up)
```
yield return new object[] { string.Join("", Enumerable.Range(1, 300).ToArray()), '1', 'x' }
```

Reply:
> to not replace all the chars in the string?

Solely for the vectorized case it doesn't matter -- it's branchless and the exact same operations are executed in that case.
But you're right that for the scalar case there's a difference, as the branch predictor will do its job.

Will put same random chars (with constant seed?) into string.

Reply:
--> 4d629c93bfd5c23082aff940e053821a0e0281ab